### ISSUE TITLE
Some rules complain due to prefix semicolon

### ISSUE THREAD

**Issue by hax** [palantir/tslint#3598]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.7.0
- __Running TSLint via__: (pick one) CLI / Node.js API / VSCode / grunt-tslint / Atom / Visual Studio / etc

#### TypeScript code being linted

```ts
let a, b
if (true) 0
;[a, b] = [1, 2]
```

with `tslint.json` configuration:

```json
"rules": {
		"semicolon": [true, "never"],
		"align": [true, "statements"],
		"curly": [true, "ignore-same-line"],
		"whitespace": [true, "check-separator"]
}
```

#### Actual behavior

ERROR: src/timer.ts[2, 1]: if statements must be braced
ERROR: src/timer.ts[3, 2]: statements are not aligned
ERROR: src/timer.ts[3, 2]: missing whitespace

#### Expected behavior

No error

**Comment by ajafff** [comment]
A statement spans from its begin until the semicolon or a newline if there is a semicolon inserted by ASI.

```ts
0 // this statement ends on the next line after the semicolon
; [a, b] // therefore this statement is not aligned with the other one
```

This is working as intended, because the rules currently don't support this code style.

Closing as a duplicate of #3201

**Comment by hax** [comment]
@ajafff  Is it possible to support prefix semicolon style? Or should I open a separate issue for this?

**Comment by ajafff** [comment]
@hax It would be possible to support this code style for the rules mentioned above. But someone has to submit a PR with the necessary changes.

Since the other issue linked above already describes this problem, I don't think we need another issue to track this.

**Comment by hax** [comment]
@ajafff I read the source code of the tests, it already include prefix style (see https://github.com/palantir/tslint/blob/master/test/rules/semicolon/never/test.ts.lint#L135-L145 ), so I'm confused.

**Comment by ajafff** [comment]
@hax That's just the semicolon rule. It doesn't care about your code style. It just looks for unnecessary / required semicolons.
The rules you mentioned above need to be updated to support this specific code style.

**Comment by zheeeng** [comment]
@hax There is a workaround:
```
let a, b
if (true) 0
;
[a, b] = [1, 2]
```
